test/awa_test_server.ml

```diff
@@ -195,29 +207,33 @@ let rec serve t cmd =
     let* _ = Driver.send_msg t msg in
     Ok ()
 
-let user_db =
-  (* User foo auths by passoword *)
-  let foo = Auth.make_user "foo" ~password:"bar" [] in
+let user_auth =
   (* User awa auths by pubkey *)
   let fd = Unix.(openfile "test/data/awa_test_rsa.pub" [O_RDONLY] 0) in
   let file_buf = Unix_cstruct.of_fd fd in
   let key = Result.get_ok (Wire.pubkey_of_openssh file_buf) in
   Unix.close fd;
-  let awa = Auth.make_user "awa" [ key ] in
-  [ foo; awa ]
+  fun user userauth ->
+  match user, userauth with
+  | "foo", Awa.Server.Password "bar" ->
+    true
+  | "awa", Awa.Server.Pubkey pubkeyauth ->
+    Awa.Server.verify_pubkeyauth ~user:"awa" pubkeyauth &&
+    Awa.Server.pubkey_of_pubkeyauth pubkeyauth = key
+  | _ -> false
```

Comment:
This shows a different way to do authentication - we could even read the public key on request.